addons/vehicleDamage/functions/fnc_processHit.sqf

```diff
@@ -0,0 +1,319 @@
+    {
+        _x setDamage 1;
+        _x setVariable [QEGVAR(medical,lastDamageSource), _injurer];
+        _x setVariable [QEGVAR(medical,lastInstigator), _injurer];
+    } forEach crew _vehicle;
+    _vehicle setDamage 1;
+    _return
+};
+
+private _projectileConfig = _projectile call CBA_fnc_getObjectConfig;
+
+private _warheadTypeStr = getText (_projectileConfig >> "warheadName");
+private _incendiary = [_projectileConfig >> QGVAR(incendiary), "NUMBER", -1] call CBA_fnc_getConfigEntry;
+private _warheadType = ["HE", "AP", "HEAT", "TandemHEAT"] find _warheadTypeStr; // numerical index for warhead type for quicker checks. Numbers defined in script_macros.hpp
+if (_warheadType < 0) then {
+    _warheadType = WARHEAD_TYPE_NONE;
+};
+if (_incendiary < 0) then {
+    _incendiary = [0.3, 0.1, 1, 1, 0] select _warheadType;
+};
+
+private _projectileExplosive = [_projectileConfig >> "explosive", "NUMBER", 0] call CBA_fnc_getConfigEntry;
+private _minDamage = [_hitpointConfig >> "minimalHit", "NUMBER", 0] call CBA_fnc_getConfigEntry;
+
+if (_warheadType isEqualTo WARHEAD_TYPE_AP) then {
+    // change damage based on projectile speed (doesn't do this in vanilla ARMA believe it or not)
+    if !(isNull _injurer) then {
+        private _airFriction = [_projectileConfig >> "airFriction", "NUMBER", 0] call CBA_fnc_getConfigEntry;
+        private _distance = _injurer distance _vehicle;
+        _newDamage = (1 - _projectileExplosive) * _newDamage * exp(_airFriction * _distance);
+    };
+};
+
+if (_newDamage < _minDamage) exitWith {
+    TRACE_4("minimum damage returning",_newDamage,_minDamage,_warheadTypeStr,_hitArea);
+    _return
+};
+
+if (_minDamage isEqualTo 0) then {
+    _minDamage = 1;
+};
+
+private _ammoEffectiveness = if (_warheadType isEqualTo WARHEAD_TYPE_AP) then {
+    0.15 max (_newDamage / 500)
+} else {
+    ((_newDamage / _minDamage) * 0.4) min 1
+};
+_incendiary = _incendiary * _ammoEffectiveness;
+
+
+private _isCar = (_vehicle isKindOf "Car" && { !(_vehicle isKindOf "Wheeled_APC_F") });
+if (_isCar) then {
+    _ammoEffectiveness = (_ammoEffectiveness + (_ammoEffectiveness * 0.5)) min 1;
+};
+
+private _injuryChance = 0;
+private _injuryCount = 0;
+switch (_warheadType) do {
+    case WARHEAD_TYPE_AP: {
+        _injuryChance = (_ammoEffectiveness * 2) min 1;
+        _injuryCount = 1 + (_ammoEffectiveness * round random 9);
+    };
+    case WARHEAD_TYPE_HE: {
+        _injuryChance = 0.1; // spalling injury chance alongside direct hit potential
+        _injuryCount = 2;
+        if (_isCar) then {
+            _injuryChance = 0.8;
+            _injuryCount = 3 max random count crew _vehicle;
+        };
+    };
+    default {
+        _injuryChance = (4 * _ammoEffectiveness) min 1;
+        _injuryCount = 2 + round random 3;
+    };
+};
+
+private _currentVehicleAmmo = [_vehicle] call EFUNC(cookoff,getVehicleAmmo);
+private _chanceOfDetonation = 0;
+private _countOfExplodableAmmo = 0;
```

Comment:
This variable is only used inside the if/then below. Move it down there to reduce defining unused variables